File: _B_pendulum/python_old/hw_a/pendulumSim.py
import matplotlib.pyplot as plt
import numpy as np
import sys
sys.path.append('..')  # add parent directory
import pendulumParam as P
from signalGenerator import signalGenerator
from pendulumAnimation import pendulumAnimation
from plotData import plotData


# instantiate reference input classes
reference = signalGenerator(amplitude=0.5, frequency=0.1)
zRef = signalGenerator(amplitude=0.5, frequency=0.1)
thetaRef = signalGenerator(amplitude=.25*np.pi, frequency=.5)
fRef = signalGenerator(amplitude=5, frequency=.5)

# instantiate the simulation plots and animation
dataPlot = plotData()
animation = pendulumAnimation()

t = P.t_start  # time starts at t_start
while t < P.t_end:  # main simulation loop
    # set variables
    r = reference.square(t)
    z = zRef.sin(t)
    theta = thetaRef.square(t)
    f = fRef.sawtooth(t)
    # update animation
    state = [z[0], theta[0], 0.0, 0.0]
    animation.drawPendulum(state)
    dataPlot.updatePlots(t, r, state, f)

    t = t + P.t_plot  # advance time by t_plot
    plt.pause(0.05)

# Keeps the program from closing until the user presses a button.
print('Press key to close')
plt.waitforbuttonpress()
plt.close()

File: _B_pendulum/unsupported_languages/matlab/hw2/pendulumSim.m
pendulumParamHW2  % load parameters

% instantiate reference input classes 
reference = signalGenerator(0.5, 0.1);
zRef = signalGenerator(0.5, 0.1);
thetaRef = signalGenerator(2*pi, 0.1);   
fRef = signalGenerator(5, 0.5);


% instantiate the data plots and animation
dataPlot = dataPlotter(P);
animation = pendulumAnimation(P);

% main simulation loop
t = P.t_start;  % time starts at t_start
while t < P.t_end  
    % set variables
    r = reference.square(t);
    z = zRef.sin(t);
    theta = thetaRef.sin(t);
    f = fRef.sawtooth(t);
    % update animation and data plot
    state = [z; theta; 0.0; 0.0];
    animation.update(state);
    dataPlot.update(t, r, state, f);
    t = t + P.t_plot;  % advance time by t_plot
    pause(0.1)
end


